src/services/bundler.ts

```diff
@@ -1,61 +1,184 @@
-import JSZip from "jszip";
+import JSZip, { JSZipObject } from "jszip";
+import toml from "toml";
 
-const MAX_FILE_SIZE = 0x10000000; // 256MB
+import { convertFiles, isFontFile, isImageFile } from "./utilities";
 
-export function validateZip(file: File): Promise<string | void> {
-  const jszip = new JSZip();
-  return new Promise((resolve, reject) => {
-    if (file.size > MAX_FILE_SIZE) {
-      reject("Your archive is too big!");
-      return;
-    }
-    jszip
-      .loadAsync(file)
-      .then((zip: JSZip) => {
-        if (zip.files["lovebrew.toml"] === undefined) {
-          reject("Missing configuration file");
-        } else {
-          resolve();
-        }
-      })
-      .catch(() => {
-        resolve("Invalid archive. Refer to the documentation.");
-      });
-  });
+export async function validateZip(file: File) {
+  const zip = await JSZip.loadAsync(file);
+
+  if (zip.files["lovebrew.toml"] === undefined) {
+    return Promise.reject("Missing configuration file.");
+  }
+
+  return Promise.resolve();
 }
 
 export interface BundlerResponse {
   message: string;
-  status: number;
   file?: Promise<Blob>;
 }
 
-export function sendZip(file: File): Promise<BundlerResponse> {
-  const body = new FormData();
-  body.append("content", file);
-  const url = `${import.meta.env.DEV ? "http://localhost:5000" : ""}/data`;
-  const request = fetch(url, {
-    method: "POST",
-    body,
-  });
-  return new Promise((resolve, reject) => {
-    request
-      .then((response) => {
-        if (response.status !== 200) {
-          response.text().then((res) => {
-            reject({ message: res, status: response.status });
-          });
+interface ConfigFile {
+  metadata: {
+    title: string;
+    author: string;
+    description: string;
+    version: string;
+    icons: Record<string, string>;
+  };
+
+  build: {
+    targets: string[];
+    source: string;
+  };
+}
+
+const extensions = {
+  ctr: "3dsx",
+  hac: "nro",
+  cafe: "wuhb",
+};
+
+export async function sendContent(archive: File): Promise<BundlerResponse> {
```

Comment:
This function is very long. Can it be divided up to smaller chunks? (e.g. `findDefinedIcons()`, `getSourceFiles()`, `convertCtrMediaFiles()`, `buildRequest()`)